File: fix_rooms.py
import firebase_admin
from firebase_admin import credentials, firestore
import os

try:
    cred = credentials.Certificate(os.path.join(os.path.dirname(__file__), 'backend', 'serviceAccountKey.json'))
    firebase_admin.initialize_app(cred)
except Exception as e:
    pass

db = firestore.client()
rooms = db.collection('public_rooms').get()
for doc in rooms:
    data = doc.to_dict()
    if data.get('status') == 'busy':
        print(f"Fixing room {doc.id}...")
        db.collection('public_rooms').document(doc.id).update({'status': 'available'})
print("Done")

File: fix_rooms.js
const admin = require('firebase-admin');
const serviceAccount = require('./backend/serviceAccountKey.json');
if (!admin.apps.length) {
    admin.initializeApp({
        credential: admin.credential.cert(serviceAccount)
    });
}
const db = admin.firestore();
(async () => {
    const snapshot = await db.collection('public_rooms').get();
    for (let doc of snapshot.docs) {
        if (doc.data().status === 'busy') {
            console.log('Fixing', doc.id);
            await doc.ref.update({ status: 'available' });
        }
    }
    console.log('Done');
    process.exit(0);
})();
